kyuubi-events/src/main/scala/org/apache/kyuubi/events/EventHandlerRegister.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kyuubi.events
+
+import org.apache.kyuubi.Logging
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.config.KyuubiConf.{ENGINE_EVENT_LOGGERS, SERVER_EVENT_LOGGERS}
+import org.apache.kyuubi.events.handler.{EventHandler, EventHandlerLoader}
+
+trait EventHandlerRegister extends Logging {
+
+  def registerEngineEventLoggers(conf: KyuubiConf): Unit = {
+    val loggers = conf.get(ENGINE_EVENT_LOGGERS)
+    register(loggers, conf)
+  }
+
+  def registerServiceEventLoggers(conf: KyuubiConf): Unit = {
+    val loggers = conf.get(SERVER_EVENT_LOGGERS)
+    register(loggers, conf)
+  }
+
+  private def register(loggers: Seq[String], conf: KyuubiConf): Unit = {
+    loggers
+      .map(EventLoggerType.withName)
+      .foreach {
+        case EventLoggerType.SPARK =>
+          loadSparkEventHandler(conf)
+            .foreach(EventBus.register)
+
+        case EventLoggerType.JSON =>
+          loadJsonEventHandler(conf)
+            .foreach(EventBus.register)
+
+        case EventLoggerType.JDBC =>
+          loadJdbcEventHandler(conf)
+            .foreach(EventBus.register)
+
+        case EventLoggerType.CUSTOM =>
+          EventHandlerLoader.loadCustom(conf)
+            .foreach(EventBus.register)
+
+        case other =>
+          warn(s"Unsupported event logger: ${other.toString}")
+      }
+  }
+
+  protected def createSparkEventHandler(kyuubiConf: KyuubiConf)
+      : Option[EventHandler[KyuubiEvent]] = {
+    Option.empty
+  }
+
+  protected def createJsonEventHandler(kyuubiConf: KyuubiConf)
+      : Option[EventHandler[KyuubiEvent]] = {
+    Option.empty
+  }
+
+  protected def createJdbcEventHandler(kyuubiConf: KyuubiConf)
+      : Option[EventHandler[KyuubiEvent]] = {
+    Option.empty
+  }
+
+  private def loadSparkEventHandler(kyuubiConf: KyuubiConf): Option[EventHandler[KyuubiEvent]] = {
```

Comment:
can we add a generic method to support arbitrary event handler ? so we do not need each method for spark/json/jdbc etc..